editor/src/components/canvas/canvas-strategies/look-for-applicable-parent.tsx

```diff
@@ -0,0 +1,298 @@
+      canvasState,
+      interactionSession,
+      strategyState.startingMetadata,
+      strategyState.startingAllElementProps,
+    )
+
+    if (result == null || result.strategies.length === 0) {
+      return defaultName
+    }
+
+    const patchedCanvasState = patchCanvasStateInteractionTargetPath(
+      canvasState,
+      result.effectiveTarget,
+    )
+
+    const fittestStrategy = calcFittestStrategy(
+      result.strategies,
+      patchedCanvasState,
+      interactionSession,
+      strategyState,
+    )
+
+    return fittestStrategy.name(canvasState, interactionSession, strategyState) + '*'
+  },
+  controlsToRender: [
+    {
+      control: DragOutlineControl,
+      key: 'ghost-outline-control',
+      show: 'visible-only-while-active',
+    },
+    {
+      control: ParentOutlines,
+      key: 'parent-outlines-control',
+      show: 'visible-only-while-active',
+    },
+    {
+      control: ParentBounds,
+      key: 'parent-bounds-control',
+      show: 'visible-only-while-active',
+    },
+  ],
+  isApplicable: (canvasState, interactionSession, metadata, allElementProps) => {
+    if (interactionSession == null) {
+      return false
+    }
+
+    const strategies = isApplicableInner(canvasState, interactionSession, metadata, allElementProps)
+    return strategies != null
+  },
+
+  fitness: (canvasState, interactionSession, strategyState) => {
+    return lookForApplicableParentStrategy.isApplicable(
+      canvasState,
+      interactionSession,
+      strategyState.startingMetadata,
+      strategyState.startingAllElementProps,
+    )
+      ? 1
+      : 0
+  },
+
+  apply: (canvasState, interactionSession, strategyState) => {
+    const result = isApplicableInner(
+      canvasState,
+      interactionSession,
+      strategyState.startingMetadata,
+      strategyState.startingAllElementProps,
+    )
+
+    if (result == null) {
+      return emptyStrategyApplicationResult
+    }
+
+    const { strategies, effectiveTarget, originalTarget } = result
+    if (strategies.length < 1) {
+      return emptyStrategyApplicationResult
+    }
+
+    const patchedCanvasState = patchCanvasStateInteractionTargetPath(canvasState, effectiveTarget)
+
+    const chosenStrategy = calcFittestStrategy(
+      result.strategies,
+      patchedCanvasState,
+      interactionSession,
+      strategyState,
+    )
+
+    const chosenStrategyApplicationResult = chosenStrategy.apply(
+      patchedCanvasState,
+      interactionSession,
+      strategyState,
+    )
+
+    return strategyApplicationResult(
+      [
+        ...chosenStrategyApplicationResult.commands,
+        highlightElementsCommand(effectiveTarget),
+        updateSelectedViews('mid-interaction', originalTarget),
+        setCursorCommand('mid-interaction', CSSCursor.MovingMagic),
+      ],
+      chosenStrategyApplicationResult.customStatePatch,
+      chosenStrategyApplicationResult.status,
+    )
+  },
+}
+
+function* elementAncestry(path: ElementPath) {
```

Comment:
wow the first generator function in our codebase!

Reply:
the reason that I included it is that in this specific case the generator function doens't add a lot of cognitive overhead (besides the yield statement), and it can bail out as soon as a suitable parent was found, no need to "materialize" all the parents in the hierarchy, although very deep hierarchies are probably an edge case